patches/0004-desktop-editor-intigration.patch

```diff
@@ -0,0 +1,209 @@
+diff --git a/app/src/lib/editors/launch.ts b/app/src/lib/editors/launch.ts
+index 1d098f384..7fd8c13bf 100644
+--- a/app/src/lib/editors/launch.ts
++++ b/app/src/lib/editors/launch.ts
+@@ -13,7 +13,13 @@ export async function launchExternalEditor(
+   editor: FoundEditor
+ ): Promise<void> {
+   const editorPath = editor.path
+-  const exists = await pathExists(editorPath)
++  let exists
++  if (process.env.FLATPAK_HOST !== 'undefined') {
++    exists = await pathExists('/var/host/run'.toString().concat(editorPath))
++  } else {
++    exists = await pathExists(editorPath)
++  }
++  console.log(exists)
+   if (!exists) {
+     const label = __DARWIN__ ? 'Preferences' : 'Options'
+     throw new ExternalEditorError(
+@@ -30,12 +36,20 @@ export async function launchExternalEditor(
+   }
+ 
+   if (editor.usesShell) {
+-    spawn(`"${editorPath}"`, [`"${fullPath}"`], { ...opts, shell: true })
++    if (process.env.FLATPAK_HOST !== 'undefined') {
++      spawn('flatpak-spawn', ['--host', `"${editorPath}"`, `"${fullPath}"`], { ...opts, shell: true })
++    } else {
++      spawn(`"${editorPath}"`, [`"${fullPath}"`], { ...opts, shell: true })
++    }
+   } else if (__DARWIN__) {
+     // In macOS we can use `open`, which will open the right executable file
+     // for us, we only need the path to the editor .app folder.
+     spawn('open', ['-a', editorPath, fullPath], opts)
+   } else {
+-    spawn(editorPath, [fullPath], opts)
++    if (process.env.FLATPAK_HOST !== 'undefined') {
```

Comment:
I have some ideas for upstream and making it easier for this `flatpak-spawn` support to not need patching like this, but first  I'd like to confirm what `process.env.FLATPAK_HOST` returns in the dev tools when you're using the test build - could you share a screenshot of that for my curiosity @advaithm ?

Reply:
Flatpak host is a env var I launch GitHub desktop with. It's value will always be 1